lighthouse-core/lib/network-request.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Fills most of the role of NetworkManager and NetworkRequest classes from DevTools.
+ * @see https://cs.chromium.org/chromium/src/third_party/blink/renderer/devtools/front_end/sdk/NetworkRequest.js
+ * @see https://cs.chromium.org/chromium/src/third_party/blink/renderer/devtools/front_end/sdk/NetworkManager.js
+ */
+
+const URL = require('./url-shim');
+const resourceTypes = require('../../third-party/devtools/ResourceType').TYPES;
+
+const SECURE_SCHEMES = ['data', 'https', 'wss', 'blob', 'chrome', 'chrome-extension', 'about'];
+
+module.exports = class NetworkRequest {
+  constructor() {
+    this.requestId = '';
+    this._requestId = '';
+    this.connectionId = '';
+    this.connectionReused = false;
+
+    this.url = '';
+    this._url = '';
+    this.protocol = '';
+    this.isSecure = false;
+    this.parsedURL = /** @type {LH.WebInspector.ParsedURL} */ ({scheme: ''});
+    this.documentURL = '';
+
+    this.startTime = -1;
+    /** @type {number} */
+    this.endTime = -1;
+    /** @type {number} */
+    this._responseReceivedTime = -1;
+
+    this.transferSize = 0;
```

Comment:
if you just went with `this._transferSize` I feel like this PR would shrink by 30%, but presumable there was a reason :)

Reply:
well transferSize is the only one we have to standardize in the middle of this PR to get everything to work thanks to the way that the old network requests worked, since we want to end up on `transferSize` in the end, makes more sense to just go straight to it :)